modules/primo/src/Primo/Ting/PrimoStatementRenderer.php

```diff
@@ -0,0 +1,191 @@
+      foreach ($rendered_group as $parameter_key => $parameter_value) {
+        if(isset($carry[$parameter_key])) {
+          $carry[$parameter_key] = array_merge($carry[$parameter_key], $parameter_value);
+        }
+        else {
+          $carry[$parameter_key] = $parameter_value;
+        }
+      }
+      return $carry;
+    }, []);
+  }
+
+  /**
+   * Verify we can process the statements and convert them to groups.
+   *
+   * @param mixed[] $statements
+   *   One or more instances of BooleanStatementGroup and/or
+   *   TingSearchFieldFilter.
+   *
+   * @return \Ting\Search\BooleanStatementGroup[]
+   *   Statements processed into a list of one or more groups.
+   *
+   * @throws \Ting\Search\UnsupportedSearchQueryException
+   *   Thrown if the statement is not supported by Primo.
+   */
+  public function preprocessStatements(array $statements) {
+    return array_reduce($statements, function ($carry, $statement) {
+      // SAL dictates that all statements added on their own will be AND'ed
+      // together and this aligns with Primo where all query= statements are
+      // AND'ed, so if we hit a statement we can just add it and continue.
+      if ($statement instanceof TingSearchFieldFilter) {
+        // Convert the statement to a group.
+        $carry[] = new BooleanStatementGroup([$statement]);
+        return $carry;
+      }
+
+      // We have a group, examine it to determine whether we Primo will be able
+      // to execute the statements.
+      /** @var \Ting\Search\BooleanStatementGroup $statement */
+      $group_statements = $statement->getStatements();
+
+      // Collect the unique field-names in the group.
+      $field_names = array_unique(array_map(function ($statement) {
+        /** @var \Ting\Search\TingSearchFieldFilter $statement */
+        return $statement->getName();
+      }, $group_statements));
+
+      // If the group is joined by anything but AND we require the field-name
+      // to be the same so that we can combine the field statements into
+      // a single "query=name,exact,value OR VALUE OR VALUE".
+      if (count($field_names) > 1) {
+        // The group references more than one field, if it joined by AND we
+        // can stil support it.
+        if ($statement->getLogicOperator() === BooleanStatementGroup::OP_AND) {
+          // Wrap each statement in the group in its own group and add it. This
+          // will work as
+          // "name=value AND (name=value AND name=value)"
+          // equals
+          // "name=value AND name=value AND name=value".
+          $carry = array_merge($carry, array_map(function ($statement) {
+            return new BooleanStatementGroup([$statement]);
+          }, $group_statements));
+        }
+        else {
+          // The group referenced more than one field and was not joined by AND.
+          throw new UnsupportedSearchQueryException("Primo only supports AND between different field.");
```

Comment:
Nitpicking field**s**.

Also: Would it be possible to provide more information about what fields conflicted or query was attempted?